pkg/server/server.go

```diff
@@ -1464,11 +1464,16 @@ func (s *Server) ListStores(ctx context.Context, req *openfgav1.ListStoresReques
 		Method:  method,
 	})
 
+	storeIDs, err := s.checkAuthzListStores(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	q := commands.NewListStoresQuery(s.datastore,
```

Comment:
Suggestion: add a note here saying that even though we have the list of store IDs, we need to call ListStoresQuery to fetch the entire metadata of the store.

This was something not immediately obvious to me  :)

(Come to think of it, `ListStoresQuery` was a poor name choice, maybe DescribeStores would have been more descriptive..)